photon-client/src/main/scala/com/linkedin/photon/ml/data/DataValidators.scala

```diff
@@ -151,4 +244,95 @@ object DataValidators extends Logging {
       throw new IllegalArgumentException(s"Data Validation failed:\n${dataErrors.mkString("\n")}")
     }
   }
+
+  /**
+   * Validate a data frame using one or more data point validators.
+   *
+   * @param dataSet The input data frame
+   * @param perSampleValidators A list of (data validator, input column name, error message) triples
+   * @return The list of validation error messages for the input data frame
+   */
+  private def validateDataFrame(
+      dataSet: DataFrame,
+      perSampleValidators: List[(((Row, String) => Boolean), InputColumnsNames.Value, String)],
+      inputColumnsNames: InputColumnsNames,
+      featureSectionKeys: Set[FeatureShardId]): Seq[String] = {
+
+    perSampleValidators
+      .flatMap { case (validator, columnName, msg) =>
+        val rows = dataSet.collect()
+        rows.flatMap{ case (r) =>
+          if (columnName.equals(InputColumnsNames.FEATURES_DEFAULT)) {
+            featureSectionKeys.map { case featureKey =>
+              val dataSetContainsColumn = dataSet.columns.contains(featureKey)
+              val result = dataSetContainsColumn match {
+                case true => {
+                  validator(r, featureKey)
+                }
+                case _ =>
+              }
+              (result, msg)
+            }
+          } else {
+            val transformedColumnName = Try(inputColumnsNames.apply(columnName))
+            val result = transformedColumnName match {
+              case Success(_) => {
+                val dataSetContainsColumn = dataSet.columns.contains(transformedColumnName.get)
+                dataSetContainsColumn match {
+                  case true => validator(r, transformedColumnName.get)
+                  case _ =>
+                }
+              }
+              case _ =>
+            }
+            Set((result, msg))
```

Comment:
As above: all of these column names must be present, since a) they're the names the user is telling us to use and b) we should be loading the `DataFrame` correctly using them.

`apply` is a [special function in Scala](https://stackoverflow.com/questions/9737352/what-is-the-apply-function-in-scala) which can be invoked using only the braces: `inputColumnsNames(columnName)`.

I think lines 277 - 288 should be replaced by:
```
(validator(r, inputColumnsNames(columnName)), msg)
```

Reply:
There are cases where "weight" and/or "offset" (and potentially "response" if we are scoring only) aren't required, so I left part of this in to check for those.